src/app/components/calendar/calendarCell.tsx

```diff
@@ -1,24 +1,27 @@
+import { CANVAS_BACKGROUNDS } from "@/app/data/configurationFormData";
 import { libreBaskerville, raleway } from "../../utils/fonts";
+import { CalendarSetupProps } from "@/app/utils/types";
+
+type CanvasValueOf<T> = T[keyof T];
 
 type Props = {
   title: string;
   dayNumber: number;
-  bgType: 'lines' | 'dots' | 'blank';
+  bgType: CanvasValueOf<Pick<CalendarSetupProps, "canvas">>;
```

Comment:
This is the same as `CalendarSetupProps['canvas']`, right?

Also, in `src/app/utils/types.ts`, you could name the type and import it.

```ts
export type CalendarCanvas = 'lines' | 'dots' | 'empty';

export type CalendarSetupProps = {
  type: "weekly" | "monthly";
  theme: 'classic' | 'minimalism';
  canvas: CalendarCanvas;
  color: 'blackAndWhite' | 'red' | 'blue' | 'green';
}
```

And similarly with the other props if you need them elsewhere